src/core/data-type.js

```diff
@@ -5,17 +5,27 @@
  * Also adds the CSS for the data type tooltip.
  * Set `conf.highlightVars = true` to enable.
  */
-import tooltipStyle from "text!../../assets/datatype.css";
+import { fetchAsset } from "./text-loader.js";
 
 export const name = "core/data-type";
 
-export function run(conf) {
+const tooltipStylePromise = loadStyle();
+
+async function loadStyle() {
```

Comment:
I see this being used multiple times. Can we move this to utils?
``` js
async function loadStyle(filename) {
  try {
    return (await import(`text!../../assets/${filename}`)).default;
  } catch {
    return fetchAsset(filename);
  }
}
// loadStyle("datatype.css");
```

Reply:
No, dynamic import with dynamic value will prevent bundler from merging assets into one file, because they are not smart enough 😂

Reply:
(╯°□°）╯︵ ┻━┻

Reply:
I hope there is a better way... but this is my current best.

Reply:
Why do we need them as dynamic import? Can't we do static import and then inject style?

Reply:
Static import will break native es module system because importing raw string is not in the spec. 🤷‍♀️

Reply:
Can we use something like webpack-dev-server for dev so that we can use ESM, but bundle them in production?

Reply:
I'm not sure how would that work?

Reply:
I've a potentially really bad idea. Convert CSS files to JS during builddeps or similar step.
``` js
export default `#i-am-css {}`
```

Reply:
Fine with the above `loadStyle()` way also though.

Reply:
Haven't thought that but should be doable. Interesting 👀

Reply:
Some objecting opinions:

* Additional code to generate css-js is not quite less than the current method
* CSS generation requires additional build method after modification, while the current method allows you to just refresh on the browser to get the latest bits (if you use ES module version to test)

Reply:
Let's stick with the present way then :+1: